python/pyspark/ml/wrapper.py

```diff
@@ -48,12 +51,27 @@ def __init__(self, java_obj: Optional["JavaObject"] = None):
         self._java_obj = java_obj
 
     def __del__(self) -> None:
-        from pyspark.core.context import SparkContext
-
-        if SparkContext._active_spark_context and self._java_obj is not None:
-            SparkContext._active_spark_context._gateway.detach(  # type: ignore[union-attr]
-                self._java_obj
-            )
+        if is_remote():
+            if self._java_obj is not None and "." not in self._java_obj:
+                try:
+                    session = SparkSession.getActiveSession()
+                    if session is not None:
+                        client = session.client
+                        req = client._execute_plan_request_with_metadata()
```

Comment:
What's this call doing?

Reply:
Looks like it creates an empty request.  It is used in other places.

Reply:
Yes, it creates a grpc request, which will be used in the line 63